packages/api/AlvTime.Persistence/Repositories/FlexhourStorage.cs

```diff
@@ -329,27 +323,31 @@ public PaidOvertimeEntry RegisterPaidOvertime(GenericHourEntry request, int user
         throw new ValidationException("Not enough available hours");
     }
 
+
+
     public PaidOvertimeEntry CancelPayout(int userId, int id)
     {
-        var payout = _context.PaidOvertime.FirstOrDefault(po => po.Id == id && po.User == userId);
+        var paidOvertime = _context.PaidOvertime.FirstOrDefault(po => po.Id == id && po.User == userId);
 
-        if (!CanBeDeleted(payout))
+        if (!CanBeDeleted(paidOvertime))
         {
             throw new ValidationException("Selected payout must be latest ordered payout");
         }
 
-        if (payout != null && payout.Date.Month >= DateTime.Now.Month && payout.Date.Year == DateTime.Now.Year)
+        if (paidOvertime != null && paidOvertime.Date.Month >= DateTime.Now.Month && paidOvertime.Date.Year == DateTime.Now.Year)
         {
-            _context.PaidOvertime.Remove(payout);
+            _context.PaidOvertime.Remove(paidOvertime);
             _context.SaveChanges();
 
+            _salaryService.DeleteOvertimePayout(userId, paidOvertime.Id);

```

Comment:
Hva skjer her dersom `_context.PaidOvertime.Remove` går gjennom, men `_salaryService.DeleteOvertimePayout` feiler? Da tror jeg det blir noe trøbbel. Kan vi vurdere å wrappe dette i en transaksjon? Se for eksempel svaret på denne posten: https://stackoverflow.com/questions/815586/using-transactions-or-savechangesfalse-and-acceptallchanges

Reply:
Må legge både registrering og sletting i transaksjon

Reply:
Fixed (lagt inn transaction på både registrering og sletting)